webapp/src/serial.tsx

```diff
@@ -0,0 +1,309 @@
+/// <reference path="../../localtypings/smoothie.d.ts" />
+
+import * as React from "react"
+import * as pkg from "./package"
+import * as core from "./core"
+import * as srceditor from "./srceditor"
+import * as sui from "./sui"
+import * as codecard from "./codecard"
+import * as canvaschart from "./canvaschart"
+
+import Cloud = pxt.Cloud
+import Util = pxt.Util
+
+const lf = Util.lf
+
+export class Editor extends srceditor.Editor {
+    charts: Chart[] = []
+    chartIdx: number = 0
+    consoleBuffer: string = ""
+    // TODO pass these values in with props or config
+    isSim: boolean = true
```

Comment:
This should be in "state"

Reply:
Post-beta TODO

Reply:
See https://github.com/Microsoft/pxt/issues/3113